File: Edges Detection/edges.py
import cv2
import numpy as np

original = cv2.imread('image.jpeg', 0)
img = cv2.imread('image.jpeg', 0)

def nothing(x):
    pass



cv2.namedWindow('Edges', cv2.WINDOW_NORMAL)
cv2.createTrackbar('Thresh Value', 'Edges', 0, 255, nothing)
cv2.createTrackbar('Min Value', 'Edges', 0, 100, nothing)
while(1):
    value = cv2.getTrackbarPos('Thresh Value', 'Edges')
    minVal = cv2.getTrackbarPos('Min Value', 'Edges')
    ret,thresh3 = cv2.threshold(img,value,255,cv2.THRESH_TRUNC)
    edges = cv2.Canny(thresh3, minVal, minVal*3)
    cv2.imshow('Edges', np.hstack((original,thresh3,edges)))
    #cv2.imshow('Trunc Threshold', thresh3)
    k = cv2.waitKey(1) & 0xFF
    if k == 27:
        break

cv2.destroyAllWindows()

'''
img = cv2.GaussianBlur(img, 5)

cv2.namedWindow('Edges', cv2.WINDOW_NORMAL)

cv2.createTrackbar('MinVal', 'Edges', 0, 300, nothing)

while(1):
    minVal = cv2.getTrackbarPos('MinVal', 'Edges')
    edges = cv2.Canny(img, minVal, minVal*3)
    cv2.imshow('Edges', edges)
    k = cv2.waitKey(1) & 0xFF
    if k == 27:
        break

cv2.destroyAllWindows()
'''

File: Edges Detection/edges.cpp
#include "opencv2/imgproc.hpp"
#include "opencv2/highgui.hpp"
#include <iostream>

using namespace cv;

Mat src, src_gray, auxiliar;
Mat dst, detected_edges;

int lowThreshold = 0;
int thresh_value = 0;
const int max_lowThreshold = 100;
const int ratio = 3;
const int kernel_size = 3;
const char* window_name = "Edge Map";

static void CannyThreshold(int, void*)
{
    threshold( src_gray, auxiliar, thresh_value, 255, 2 );
    blur(auxiliar, detected_edges, Size(3,3));
    Canny(detected_edges, detected_edges, lowThreshold, lowThreshold*ratio, kernel_size);
    dst = Scalar::all(0);
    src.copyTo(dst, detected_edges);
    namedWindow(window_name, WINDOW_NORMAL);
    imshow(window_name, detected_edges);
}

int main(int argc, char** argv)
{
    src = imread(argv[1], 1);
    dst.create(src.size(), src.type());
    cvtColor(src, src_gray, COLOR_BGR2GRAY);
    namedWindow(window_name, WINDOW_NORMAL);
    createTrackbar("Threshold Value: ", window_name, &thresh_value, 255, CannyThreshold);
    createTrackbar("Canny Min Value: ", window_name, &lowThreshold, max_lowThreshold, CannyThreshold);
    CannyThreshold(0,0);
    waitKey(0);
    return 0;
}